packages/lib-classifier/src/plugins/tasks/TextTask/components/TextTask/TextTask.spec.js

```diff
@@ -27,131 +27,253 @@ describe('TextTask', function () {
+        expect(annotation.update.withArgs('This has been edited.')).to.have.been.calledOnce()
+      })
     })
 
-    it('should preserve an existing annotation', function () {
-      const textArea = wrapper.find(TextArea).getDOMNode()
-      expect(textArea.value).to.equal('Hello, this is an existing annotation')
+    describe('on unmount', function () {
+      before(function () {
+        sinon.spy(annotation, 'update')
+        annotation.update('Hello, this is an existing annotation')
+        task.setAnnotation(annotation)
+        wrapper = mount(
+          <TextTask
+            task={task}
+          />
+        )
+        const textArea = wrapper.find(TextArea).getDOMNode()
+        textArea.value = 'This has been edited.'
+        wrapper.unmount()
+      })
+
+      after(function () {
+        annotation.update.restore()
+      })
+
+      it('should save the current text', function () {
+        expect(annotation.update.withArgs('This has been edited.')).to.have.been.calledOnce()
+      })
     })
   })
-
-  describe('on change', function () {
+  
+  describe('task with suggestions', function () {
+    let wrapper
+    const suggestions = [
+      'a transcribed sentence',
+      'a transcribed sentience',
+      'a transribed sentence',
+      'a conscripted sentience'
+    ]
     before(function () {
-      sinon.spy(annotation, 'update')
-      annotation.update('Hello, this is an existing annotation')
-      task.setAnnotation(annotation)
-      wrapper = mount(
+      wrapper = shallow(
         <TextTask
+          suggestions={suggestions}
           task={task}
         />
       )
-      const textArea = wrapper.find(TextArea).getDOMNode()
-      textArea.value = 'This has been edited.'
-      wrapper.find(TextArea).simulate('change')
     })
 
-    after(function () {
-      annotation.update.restore()
+    it('should render a TextTaskWithSuggestions', function () {
+      expect(wrapper.find(TextTaskWithSuggestions)).to.have.lengthOf(1)
     })
 
-    it('should save the current text', function () {
-      expect(annotation.update.withArgs('This has been edited.')).to.have.been.calledOnce()
+    describe('text tagging', function () {
+      beforeEach(function () {
+        sinon.spy(annotation, 'update')
+        annotation.update('Hello, this is some test text.')
+        wrapper = mount(
+          <TextTask
+            suggestions={suggestions}
+            task={task}
+          />
+        )
+        const insertionButton = wrapper.find('button').find({ value: 'insertion' })
+        const fakeEvent = {
+          currentTarget: {
+            value: 'insertion'
+          }
+        }
+        const textInput = wrapper.find('input').getDOMNode()
+
+        textInput.selectionStart = 7
+        textInput.selectionEnd = 11
+        insertionButton.simulate('click', fakeEvent)
+      })
+
+      afterEach(function () {
+        annotation.update.restore()
+      })
+
+      it('should tag text', function () {
```

Comment:
These tests are repeated to make sure the TextTagButtons are updating the text and annotation as expected per this component's `setTagSelection` method between the two very similar though in some ways different DefaultTexTask and TextTaskWithSuggestions components. The DefaultTextTask tested in the initial tests uses a `textarea` while the TextTaskWithSuggestions uses an `input`, and while I can't think of related different behaviors between the two components off the top of my head, I think it's good to test both separately.

Reply:
We're testing the same behavior for either text task component. Both should be able to update annotations, insert text tags, etc. This could be refactored to have a single set of functions called for each test block, but tbh, I prefer the explicitness in reading through the tests even if they're repeated.